docs/decisions/0003-lightweight-extension-points.rst

```diff
@@ -0,0 +1,202 @@
+Lightweight extension points
+############################
+
+Status
+******
+
+**Draft** *2024-02-27*
+
+Context
```

Comment:
I think the context described here is focusing too much on the particular integration that we are looking to build on top of an extension mechanism. Given the experience we have had trying to modify ORA2 so far the issue is that the needs of different users like edx.org, the spanish universitites, other on campus intiatives and other members of the community don't align.

I propose we focus the ADR towards the issues created by not having any extension mechanism that does not require forking the code. By this I mean, ORA is in itself designed to be extended, it is possible to create new steps and configure them in, however this requires essencially modifiying the core of the ORA repo a.ka. forking it.

We have at least two examples one being the code we wrote for the spanish campus uses and the other being RG's ai-ora2 work.

I propose something like this for the context:


>   Open Responses are commonly used in education for assessment purposes and the flexibility of ORA has made it a key feature of the Open edX platform. However as developers it is very hard to accomodate the needs of educators and other stakeholders in the teaching process when they want to enhance the learners experience with other tools such as plagiarism detection, AI grading, coding graders and others.
> 
>   As the code is open it is always possible to modify it incurring in a technical mainteinance debt that for the most part either prevents educators from actually experimenting or prevents the instances to upgrade to newer versions once the changes to ORA become important in the teaching process.

Reply:
I see. I already included the suggestions. Thanks for the feedback!